pkg/reconciler/subscription/subscription_test.go

```diff
@@ -58,17 +60,13 @@ const (
 // subscriptions have: channel -> SUB -> subscriber -viaSub-> reply
 
 var (
-	channelDNS = "channel.mynamespace.svc." + utils.GetClusterDomainName()
-	channelURI = "http://" + channelDNS
-
+	channelDNS    = "channel.mynamespace.svc." + utils.GetClusterDomainName()
 	subscriberDNS = "subscriber.mynamespace.svc." + utils.GetClusterDomainName()
-	subscriberURI = "http://" + subscriberDNS
-
-	replyDNS = "reply.mynamespace.svc." + utils.GetClusterDomainName()
-	replyURI = "http://" + replyDNS
-
-	serviceDNS = serviceName + "." + testNS + ".svc." + utils.GetClusterDomainName()
-	serviceURI = "http://" + serviceDNS
+	subscriberURI = &apis.URL{Scheme: "http", Host: subscriberDNS}
+	replyDNS      = "reply.mynamespace.svc." + utils.GetClusterDomainName()
+	replyURI      = &apis.URL{Host: replyDNS, Scheme: "http"}
+	serviceDNS    = serviceName + "." + testNS + ".svc." + utils.GetClusterDomainName()
+	serviceURI    = &apis.URL{Host: serviceDNS, Scheme: "http", Path: "/"}
```

Comment:
(Copying a comment that got lost in a force-push):

The existing path didn't set Path, and subscriberURI and replyURI methods also don't...

Reply:
There's something really weird happening over here which I'm not able to figure out. so i removed the *Path* parameter and as expected, the unit tests started to fail. so i debugged through the source and found that in subscription reconciler we get the urls via this stupid statement:  https://github.com/knative/eventing/blob/530a936c823f918cd68d3ee3a8e1ba2f523fd581/pkg/reconciler/subscription/subscription.go#L183
which adds the path to url object for a k8s service from here : https://github.com/knative/eventing/blob/37a2d83d5a75ec787cf6f0bf618e5f5344faa88c/pkg/duck/subscriber.go#L105

but is I change this, Integration tests simply falls apart. I'd appreciate some direction over here thanks

Reply:
The tracing tests are looking for the "/" in the request here:

https://github.com/knative/eventing/blob/master/test/conformance/helpers/channel_tracing_test_helper.go#L163

But get a span like this:

            "Span": {
              "timestamp": 1569928956226906,
              "duration": 18470,
              "traceId": "e13e18ada1cd728457454f18108966b5",
              "id": "2b61b8c04ce93eab",
              "parentId": "41eef8ebdaebfb28",
              "kind": "CLIENT",
              "localEndpoint": {
                "serviceName": "imc-dispatcher",
                "ipv4": "10.15.252.156",
                "port": 80
              },
              "tags": {
                "http.host": "",
                "http.method": "POST",
                "http.path": "",
                "http.status_code": "202",
                "http.url": "http://logger.includes-incoming-trace-id-in-memory-channel-dlgsl.svc.cluster.local",
                "opencensus.status_description": "OK"
              }
            },

IMO, I would remove the "path" qualification from that tracing test; it doesn't seem helpful, but does seem fragile.

Reply:
@Harwayne who wrote the tracing tests.

Reply:
Sure, go ahead and remove any of the paths that are just '/'.